Assets/Code/src/Runtime/Match/MatchSimulation.cs

```diff
@@ -39,6 +44,14 @@ public class MatchSimulation : IMatchSimulation {
     }
   }
 
-}
+  void SimulateRestrict(ref MatchState state, MatchInputContext input, params Type[] restrictions) {
+	 foreach (var SimComponent in SimulationComponents) {
+		if (restrictions.Contains(SimComponent.GetType()))
```

Comment:
Calling GetType and comparing against types can get very expensive and may allocate memory. 

Suggested fix: Create separate IMatchSimulation[] arrays for each MatchStateID, put them into a Dictionary<MatchStateID, IMatchSimulation[]> and retrieve the simulation set via state.StateID, then simulate. Uses marginally more permanent memory for less GC allocation and faster execution.